### PR TITLE
Modify routes to pass auth info via http headers  

### PR DESCRIPTION
**Pull request by geoff-reid**
@keedya @tannoa2 @uppalk1 @RackHD/corecommitters 

This PR modifies all routes to use http headers instead of the query string to pass ucs manger auth info.  The service was also modified to use https instead of http to encrypt the new auth headers.

### CODE REVIEW

**Comment by yyscamper** [comment]
For only simple access control, you could use basic authentication: https://en.wikipedia.org/wiki/Basic_access_authentication

File: controllers/ucs_controller.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                for x in computeRackUnit:
+                    server = {}
+                    server["name"] = x.rn
+                    server["path"] = x.dn
+                    server["macs"] = []
+                    try:
+                        macs = handle.query_children(in_dn=x.dn, class_id='PciEquipSlot')
+                    except UcsException as e:
+                        handle.logout()
+                        return 'Inrternal Server Error', e.error_descr, 500
```
Comment (anhou) [inline controllers/ucs_controller.py]: typo

**Comment by geoff-reid** [comment]
@yyscamper please review changes to use http headers instead of POST routes.

File: controllers/ucs_controller.py [hunk 2]
```diff
@@ -221,7 +240,10 @@ def getServiceProfile(host=None, user=None, password=None):
 
 
 @http_body_factory
-def powerMgmt(host=None, user=None, password=None, identifier=None, action=None):
+def powerMgmt(identifier=None, action=None):
+    host = request.headers.get('ucs-host')
+    user = request.headers.get('ucs-user')
+    password = request.headers.get('ucs-password')
```
Comment (yyscamper) [inline controllers/ucs_controller.py]: Considering there is a lot copies for above 3 lines of code, create a function for them will remove these duplicated code and conform to the clean code practice: https://github.com/ryanmcdermott/clean-code-javascript#remove-duplicate-code

**Comment by geoff-reid** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Comment by geoff-reid** [comment]
test this please

**Pull request merged**